### DISCUSSION TITLE
Is sanitize expected to ignore camel-cased attributes even when set in the schema?

### DISCUSSION THREAD

**Discussion by xiaohk** [rehypejs/rehype#183] [Q&A] [answered]
Hello, I'm wondering if it's expected for camel-cased attributes to be removed even when I include them in the schema?

See [this demo](https://stackblitz.com/edit/vitejs-vite-exevin?file=src%2FApp.tsx) for an example.

Input HTML:
```html
<div class="my-p-class" my-name="cool" coolName="very cool"> Hello World!</div>
```

Output HTML:
```html
<div class="my-p-class" my-name="cool"> Hello World!</div>
```

Schema Setting:
```js
  const customSchema = {
    ...defaultSchema,
    attributes: {
      ...defaultSchema.attributes,
      '*': ['className', 'my-name', 'coolName'],
    },
  };
```

**Comment by ChristianMurphy** [comment, marked as answer]
Hey @xiaohk!

> Hello, I'm wondering if it's expected for camel-cased attributes to be removed even when I include them in the schema?

HTML is case insensitive (lowercase), and the schema does not rename attributes

The custom schema you are probably looking for is

```ts
  const customSchema = {
    ...defaultSchema,
    attributes: {
      ...defaultSchema.attributes,
      '*': ['class', 'my-name', 'coolname'],
    },
  };
```

That said with all the camel case, I get the sense you may actually may want XML or JSX? Are you interacting with a domain specific language or templating language?

**Reply by xiaohk** [reply]
Oh I see!! Yes, changing `coolName` to `coolname` in `customSchema` fixed the problem. Thank you very much!

I need these custom attributes for sanitizing custom elements ٩( ᐛ )و

**Reply by ChristianMurphy** [reply]
> sanitizing custom elements

Okay.
HTML is case insensitive.
The component names and attributes should be lower case dash separated by conventional.

Something is off if they are case sensitive camel case.
